operator/roles/kiali-deploy/templates/openshift/deployment.yaml

```diff
@@ -39,6 +39,11 @@ spec:
         - "/kiali-configuration/config.yaml"
         - "-v"
         - "{{ kiali_vars.deployment.verbose_mode }}"
+{% if kiali_vars.deployment.system_test_codecov_enabled %}
```

Comment:
all of this is going in our operator, too?

Reply:
From the looks of this PR, this is the crux of the change - you need to pass additional command line options to the kiali binary - is that all you need?

What I suggest you do is - reject all of the changes you do here to operator/* files, and as part of your testing, when you deploy Kiali, just edit its deployment (add these options to it) and then redeploy the pod. So, for example, in your system test procedures, after you run the operator and deploy kiali, then run something like this:

```
oc patch deployment kiali -n istio-system -p '{"spec":{"containers": {[ "command": [-/opt/kiail, -config, /kiali-config/config.yaml, -systemTest, -test.coverprofile, /opt/kiali/console/coverage.cov]]}}}' --type=merge
```

That's not exact - I didn't get the syntax perfect, but you should get the idea. Just use "oc patch" to change the command line options you want the pod to run with that you need for code coverage.

That puts the changes where they belong (in the internal system test workflow/processes) and does not encumber the kiali release with all of this extra system test/code coverage stuff that will never be used in production by users.

Reply:
OK, remove all of the changes to all of these operator/* files, and instead, your system test setup scripts need only run this one command:

```
oc patch deployment kiali --type=merge -n istio-system -p '{"spec":{"template": {"spec": {"containers": [{"name": "kiali", "image": "kiali/kiali:dev", "command": ["/opt/kiali/kiali", "-config", "/kiali-configuration/config.yaml", "-v", "4", "-systemTest", "-test.coverprofile", "/opt/kiali/console/coverage.cov"] }]}}}}'
```

I just tried it, and that does change the deployment properly. Automatically OpenShift/Kubernetes will redeploy the pod with those new options - you don't have to do it yourself.